core/src/App.html

```diff
@@ -295,181 +295,189 @@
           }
         });
         window.addEventListener('message', async e => {
-          if ('luigi.get-context' === e.data.msg) {
-            if (
-              this.get().modalIframe &&
-              e.source === this.get().modalIframe.contentWindow
-            ) {
-              let ctx = this.get().modalIframeData.context;
-              const modalConfig = {
-                ...config,
-                iframe: this.get().modalIframe,
-                context: ctx,
-                pathParams: this.get().modalIframeData.pathParams,
-                nodeParams: this.get().modalIframeData.nodeParams,
-                modal: true
-              };
-              sendContextToClient(this, modalConfig, {});
-            } else if (config.iframe) {
-              sendContextToClient(this, config, {});
-
-              const loadingIndicatorAutoHideEnabled =
-                GenericHelpers.getConfigValueFromObject(
-                  this.get(),
-                  'currentNode.loadingIndicator.hideAutomatically'
-                ) !== false;
-              if (loadingIndicatorAutoHideEnabled) {
-                this.set({ showLoadingIndicator: false });
-              }
-            }
-          }
+          this.set({url: e.origin})
+          const allIframes = Iframe.getAllIframes(node);
+          if (allIframes && allIframes.length > 0) {
```

Comment:
I would suggest avoiding more nesting by doing the following:
```
 if (!allIframes || allIframes.length === 0) { return }
// here code currently inside the if block
```

Reply:
Additionally, I would enforce  `Iframe.getAllIframes` to return always an array to make checks easier: `if(allIframes.length > 0)`

Reply:
I assume the forEach block will find zero or one Iframe that will match. Therefore I would use `some` instead and keep the code outside of the loop to avoid another level of nesting. 
Following up on the last 2 comments, I would refactor it to sth like that:
```
const validSource = allIframes.some(() => e.source === iframe.contentWindow || (this.get().modalIframe && e.source === this.get().modalIframe.contentWindow))
if (!validSource) { return }
// here code currently inside the foreach block
```

Reply:
I saw the following piece of code used 3 times, maybe we can create a variable out of it:
`this.get().modalIframe && e.source === this.get().modalIframe.contentWindow`

Reply:
Regarding the 1st and 3rd comment - after `if` block there is still one function that needs to be executed, so returning here wouldn't make sense. I left it in this `if` block.

Reply:
Correct me if I am wrong, but I think something like that should work:
```
          const isOriginWindow = (iframe) => e.source === iframe.contentWindow;
          const isOriginModal = () => this.get().modalIframe && e.source === this.get().modalIframe.contentWindow;

          const iframe = Iframe.getAllIframes(node)
            .find((iframe) => isOriginWindow(iframe) || isOriginModal());

          if (!iframe) { return };
          // Remaining code
```
Benefits of this:
- more reusability of code than with `isTheSameOrigin()`
- two levels less of indentation for 100+ lines of code ;)